pkg/cloud/gcp/actuators/machine/actuator_test.go

```diff
@@ -230,13 +230,18 @@ func TestActuatorEvents(t *testing.T) {
 						Value: providerSpec,
 					},
 				}}
-
-			// Create the machine
-			gs.Expect(k8sClient.Create(context.Background(), machine)).To(Succeed())
+			eventList := &v1.EventList{}
+			// Make sure the machine and its event are deleted when the test ends
 			defer func() {
 				gs.Expect(k8sClient.Delete(context.Background(), machine)).To(Succeed())
+				for i := range eventList.Items {
+					gs.Expect(k8sClient.Delete(context.Background(), &eventList.Items[i])).To(Succeed())
+				}
```

Comment:
To avoid any sharing of the eventList, and to make sure we delete events even when there's an earlier failure, why not do a list in here to make sure we have an up to date list
```suggestion
				
				eventList := &v1.EventList{}
				gs.Expect(k8sClient.List(context.Background(), eventList).To(Succeed())
				for i := range eventList.Items {
					gs.Expect(k8sClient.Delete(context.Background(), &eventList.Items[i])).To(Succeed())
				}
```

Reply:
Yes, makes sense to me! thanks for the suggestion.